resources/js/chat/chat-state-store.ts

```diff
@@ -96,15 +109,57 @@ export default class ChatStateStore implements DispatchListener {
         }
 
         runInAction(() => {
-          // TODO: use selectChannel?
-          if (typeof this.selected === 'number') {
-            this.channelStore.loadChannel(this.selected);
-          }
+          this.selectedChannel?.load();
 
           this.isReady = true;
         });
       }
     });
+
+    autorun(() => {
+      if (this.selected === 'join' && this.publicChannels.channels == null && !this.publicChannels.error) {
+        this.publicChannels.load();
+      }
+    });
+  }
+
+  // Only up to one join/create channel operation should be allowed to be running at any time.
+  // For consistency, the operation is considered complete when the websocket message arrives, not when the request completes.
+  // TODO: allow user cancelling operation from UI?
+  @action
+  async addChannel(channelId?: number) {
+    if (this.isAddingChannel) return;
+
+    try {
+      if (channelId != null) {
+        if (this.channelStore.channels.has(channelId)) {
+          this.selectChannel(channelId);
+          return;
+        }
+
+        showImmediateLoadingOverlay();
+
+        this.waitAddChannelId = channelId;
+        await joinChannel(channelId, core.currentUserOrFail.id);
```

Comment:
fun behavior 🤔 

0. add sleep to `ChannelsController#join` to emulate slow connection
1. go to a page
2. navigate to chat
3. navigate to join channel
4. join something
5. go back (browser) (should go to whatever previous channel was)
6. wait
7. after a while it switches channel (maybe expected but seems a bit rude?)

Or instead of wait at 6:

6. go back again (to whatever initial page was)
7. wait backend sleep to finish
8. go forward (to chat page)
9. select join channel
10. try joining any channel
11. nothing happens (not sure what happens here)